core/services/directrequestocr/drlistener.go

```diff
@@ -290,3 +291,34 @@ func (l *DRListener) markLogConsumed(lb log.Broadcast, qopts ...pg.QOpt) {
 func formatRequestId(requestId [32]byte) string {
 	return fmt.Sprintf("0x%x", requestId)
 }
+
+func (l *DRListener) timeoutRequests() {
+	defer l.shutdownWaitGroup.Done()
+	timeoutSec, freqSec, batchSize := l.pluginConfig.RequestTimeoutSec, l.pluginConfig.RequestTimeoutCheckFrequencySec, l.pluginConfig.RequestTimeoutBatchLookupSize
+	if timeoutSec == 0 || freqSec == 0 || batchSize == 0 {
+		l.logger.Warnw("request timeout checker not configured - disabling it")
+		return
+	}
+	ticker := time.NewTicker(time.Duration(freqSec) * time.Second)
+	for {
+		select {
+		case <-l.chStop:
+			ticker.Stop()
```

Comment:
A minor thing, but better to `defer ticker.Stop()` right after creation. Somebody else may miss this in a future and add a return in a different place and likely forget to call `Stop()`.